compiler/llvm/clangUtil.cpp

```diff
@@ -4022,6 +4032,73 @@ static std::string getLibraryOutputPath();
 static void moveGeneratedLibraryFile(const char* tmpbinname);
 static void moveResultFromTmp(const char* resultName, const char* tmpbinname);
 
+static llvm::CodeGenFileType getCodeGenFileType() {
+  switch (getGpuCodegenType()) {
+    case GpuCodegenType::GPU_CG_AMD:
+      return llvm::CodeGenFileType::CGFT_ObjectFile;
+    case GpuCodegenType::GPU_CG_CUDA:
+    default:
+      return llvm::CodeGenFileType::CGFT_AssemblyFile;
+  }
+}
+
+static void makeBinaryLLVMForCUDA(const std::string& artifactFilename,
```

Comment:
Thank you for factoring this out.

Nit: `makeBinaryLLVMForCUDA` -> `makeBinaryLlvmForCuda`

Reply:
It was named this way for consistency with `makeBinaryLLVM`, which is itself consistent with other `doSomethingLLVM` functions in `clangUtil.{cpp,h}`. Should I still rename?

Reply:
I see, I guess I'd keep it "as is" then.